lib/rules/prefer-expect-query-by.js

```diff
@@ -0,0 +1,99 @@
+'use strict';
+
+const { getDocsUrl } = require('../utils');
+
+const AST_NODE_TYPES = {
+  Identifier: 'Identifier',
+  MemberExpression: 'MemberExpression',
+};
+
+function isIdentifier(node) {
+  return node.type === AST_NODE_TYPES.Identifier;
+}
+
+function isMemberExpression(node) {
+  return node.type === AST_NODE_TYPES.MemberExpression;
+}
+
+function isUsingWrongQueries(node) {
+  return node.name.startsWith('getBy') || node.name.startsWith('getAllBy');
+}
+
+function isNotNullOrUndefined(input) {
+  return input != null;
+}
+
+function mapNodesForWrongGetByQuery(node) {
+  const nodeArguments = node.arguments;
+  return nodeArguments
+    .map(arg => {
+      if (!arg.callee) {
+        return null;
+      }
+      // Example: `expect(rendered.getBy*)`
```

Comment:
I made it more clear in the doc what this is. Basically it's when you don't destructure the render result:

```js
const { queryByTest } = render(<App />);

// or

const rendered = render(<App />);
```

Reply:
Ah ok, but you can't assume it will be called `rendered` as it's a var defined by the user (I use to call it `utils` actually), so I'm afraid you'll have to check this in a more generic way.

Reply:
The name of the variable does not matter. You can put whatever you want.

Reply:
Aaah my bad, reading the comment I thought it was hardcoded just for `rendered` but checking the actual code I see is generic.